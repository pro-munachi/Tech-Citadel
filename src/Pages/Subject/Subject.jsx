import { Box, Text, useColorModeValue } from "@chakra-ui/react";
import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { useParams } from "react-router-dom";
import Navbar from "../../components/Navbar/Navbar";
import { subjectInfo } from "../../state/actions/subjectAction";

const Subject = () => {
  const dispatch = useDispatch();

  const state = useSelector((state) => state.login);
  const { userInfo } = state;

  const subject = useSelector((state) => state.subjects);
  const { data } = subject;

  console.log(data);

  const { id } = useParams();

  const user = JSON.parse(localStorage.getItem("userInformations"));

  React.useEffect(() => {
    getData();
  }, []);

  const getData = () => {
    dispatch(subjectInfo(userInfo?.uid || user?.uid, id));
  };
  return (
    <Box>
      <Navbar />

      <Box
        w="80%"
        mx="auto"
        mt="30px"
        bg={useColorModeValue("lightMode.mainBgColor", "darkMode.mainBgColor")}
        p="20px"
      >
        <Box>
          <Text>Subject: {data?.name}</Text>
          <Text>Teacher: {data?.teacher}</Text>
          <Text>Assessment: {data?.assessment}</Text>
          <Text>Exam: {data?.exam}</Text>
        </Box>
      </Box>
    </Box>
  );
};

export default Subject;
