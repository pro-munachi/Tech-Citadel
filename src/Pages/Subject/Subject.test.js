import { render } from "@testing-library/react";
import Subject from "./Subject";
import { BrowserRouter } from "react-router-dom";
import { ChakraProvider, theme } from "@chakra-ui/react";
import store from "../../state/store";
import { Provider } from "react-redux";

const Wrapper = ({ children }) => {
  <ChakraProvider theme={theme}>
    <Provider store={store}>
      <BrowserRouter>{children}</BrowserRouter>
    </Provider>
  </ChakraProvider>;
};

test("Test Subject", async () => {
  const { asFragment } = render(
    <Wrapper>
      <Subject />
    </Wrapper>
  );

  expect(asFragment()).toMatchSnapshot();
});
