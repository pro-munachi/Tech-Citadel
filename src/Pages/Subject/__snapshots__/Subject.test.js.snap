// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Subject 1`] = `<DocumentFragment />`;
